object_generators/tokenizer_generator.py

```diff
@@ -0,0 +1,52 @@
+import json
+from glob import glob
+
+import streamlit as st
+
+from midi_tokenizers.midi_tokenizer import MidiTokenizer
+from midi_trainable_tokenizers.bpe_tokenizer import BpeMidiTokenizer
+from object_generators.base_tokenizer_generator import TokenizerFactory, BaseTokenizerGenerator
+
+
+class BpeMidiTokenizerFactory(TokenizerFactory):
+    tokenizer_desc = """
+    This tokenizer can be trained on tokens generated by on of the no-loss models,
+    which is passed to it by base_tokenizer parameter.
+
+    WARNING: be sure to choose base-tokenizer that was used during training if using pre-trained BPE tokenizer.
+    (I will think of a better serialization technique later)
+
+    You can train new tokenizers by messing with scripts/train_bpe.py
+    """
+
+    @staticmethod
+    def select_parameters() -> dict:
+        trained_tokenizers_options = glob("dumps/*.json")
```

Comment:
`"dumps/*.json"` is not very specific, if you need some kind of persistence please try to be more specific, e.g. "dumps/tokenizers/*.json"